coffeescript-loader/test.js

```diff
@@ -24,12 +22,7 @@ child.stderr.on('data', (data) => {
 });
 
 child.on('close', (code, signal) => {
-  ok(stdout.includes('Hello from CoffeeScript', 'Main entry transpiles'));
+  ok(stdout.includes('Hello from CoffeeScript'), 'Main entry transpiles');
   ok(stdout.includes('HELLO FROM ESM'), 'ESM import transpiles');
-
-  // There is a known issue between ESM + CJS where CJS named exports get lost:
-  // require.extentions appropriately supplies (transformed) source, but an
-  // empty object is returned for the module's exports.
-  // the import does NOT hit ESMLoader or its CJS strategy.
-  ok(!stdout.includes('Hello from CommonJS!'), 'Named CommonJS import fails');
+  ok(stdout.includes('Hello from CommonJS!'), 'Named CommonJS import fails');
```

Comment:
Does this pass now?

Reply:
It does

Reply:
And what did the trick was adding `module._compile`, it looks like?

Reply:
Yes adding `module._compile` was the critical piece. `coffeescript/register` called it in their implementation so I looked to see what was different here.